### PR TITLE
Tperps 735 ome state restoration

### PR DESCRIPTION
**Pull request by chrdevmar**
# Motivation
- In case of an OME crash or restart, it should be able to restore the state of the orderbook(s)

# Changes
- Upon startup, will request a list of known markets from the `known_markets_url` endpoint (new cli argument)
- Will then request the orderbook state for each of the known markets from the `external_book_url` endpoint (new cli argument)
- Will initialize the existing orderbooks before starting up
- new API functionality is implemented here: https://github.com/mycelium-ethereum/perpetual-api/pull/11

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,10 +21,10 @@ To run the OME, with the executioner running locally, use
 
 ## ENV Variables
 The OME supports the following ENV variables
-- executioner_address: The IP address of the executioner instance
+- known_markets_url: an endpoint which will return a list of known markets upon a GET request
+- external_book_url: an endpoint which will return an external book upon a GET request. marketId will be concatenated to this url
```
Comment (sporejack) [inline README.md:25]:
```suggestion
- `known_markets_url`: an endpoint which will return a list of known markets upon a GET request
- `external_book_url`: an endpoint which will return an external book upon a GET request. `marketId` will be concatenated to this url
```

File: src/rpc.rs [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    };
+
+    let book: ExternalBookResponse = response.json().await?;
+
+    info!("{} said {:?}", endpoint, book);
+
+    Ok(book.data)
+}
+
+pub async fn get_external_boo_k(
```
Comment (chrdevmar) [inline src/rpc.rs]: this function name has a bad `_`

Comment (sporejack) [inline src/rpc.rs]:
```suggestion
pub async fn get_external_book(
```

**Review by sporejack — CHANGES_REQUESTED** [review]
A few (small) changes necessary, but otherwise this is a really good changset.

One thing to consider is how necessary the `state` and `util` modules are now that you've tidied things up?

**Comment by chrdevmar** [comment]
> A few (small) changes necessary, but otherwise this is a really good changset.
> 
> One thing to consider is how necessary the `state` and `util` modules are now that you've tidied things up?

Have changed the small stuff. 

Regarding those two modules, I had a quick look and I'm hesitant to do anything about them in this PR. Keen to note it down as addressable tech debt though

**Review by sporejack — APPROVED** [review]

**Review by raymogg — APPROVED** [review]

**Pull request merged**